src/webgpu/api/validation/render_pipeline/fragment_state.spec.ts

```diff
@@ -49,9 +52,57 @@ g.test('color_target_exists')
     t.doCreateRenderPipelineTest(isAsync, false, badDescriptor);
   });
 
+g.test('targets_format_is_color_format')
+  .desc(
+    `Tests that color target state format must be a color format, regardless of how the
+    fragment shader writes to it.`
+  )
+  .params(u => {
+    assert(kColorTextureFormats.length + kDepthStencilFormats.length === kAllTextureFormats.length);
```

Comment:
nit. Does it make sense to put this here or as like an assert in the format table file?

Reply:
I think it makes sense for it to be here because generally most tests don't care. This one just needs to make sure that it's testing every non-color format.

But actually I can just do this in a more constructive way: start with all formats, then filter out the color formats.